lib/methods/bulk-action.ts

```diff
@@ -0,0 +1,102 @@
+import delay from 'delay'
+import {
+  BulkAction,
+  BulkActionPayload,
+  BulkActionProps,
+  BulkActionStatus,
+} from '../entities/bulk-action'
+import { PlainClientAPI } from '../plain/common-types'
+
+const DEFAULT_MAX_RETRIES = 30
+const DEFAULT_INITIAL_DELAY_MS = 1000
+const DEFAULT_RETRY_INTERVAL_MS = 2000
+
+export class BulkActionProcessingError extends Error {
```

Comment:
I'm assuming we don't have a base ContentfulError in this client package to extend from?

Reply:
yep, actually this is the 2nd error in the whole package. Mostly, the adapter throws the `id` of the error response as an error. We might define one error base class once another entity gets added (releases for example)